src/migration/utilities.js

```diff
@@ -1,23 +1,33 @@
-function initializeContract(path, deployer, accounts) {
-  const Contract = require("truffle-contract")
+function initializeContract(path, deployer, account) {
+  const Contract = require("@truffle/contract")
 
   const contract = Contract(require(path))
   contract.setProvider(deployer.provider)
   contract.setNetwork(deployer.network_id)
   // For some reason the automatic value calculation is not working, hence we do:
   contract.defaults({
-    from: accounts[0],
+    from: account,
     gas: 6.5e6,
   })
   return contract
 }
 
-function getDependency(artifacts, network, deployer, account, path) {
+async function getDeployedDependency(artifacts, deployer, account, path) {
+  let contract
+  // The following logic tries to get the deployed dependency from the local build folder, and falls back to
+  // npm imports. This makes sure that the script can be used within dex-contracts and from external projects.
+  try {
+    contract = artifacts.require(path.split("/").pop())
```

Comment:
I'd expect this to be Contract and then `let contract = await Contract.deployed()`. Why are we running deployed but not assigning the result to the variable we return?

> MyContract.deployed()
Creates an instance of the contract abstraction representing the contract at its deployed address. The deployed address is a special value given to truffle-contract that, when set, saves the address internally so that the deployed address can be inferred from the given Ethereum network being used. This allows you to write code referring to a specific deployed contract without having to manage those addresses yourself. Like at(), deployed() is thenable, and will resolve to a contract abstraction instance representing the deployed contract after ensuring that code exists at that location and that that address exists on the network being used.

(https://github.com/trufflesuite/trufflesuite.com/blob/master/src/docs/truffle/reference/contract-abstractions.md)

Reply:
No, I named the function getDeployedDependency, so I thought everyone would expect a deployed contract instance

Reply:
Here is the code: https://github.com/trufflesuite/truffle/blob/develop/packages/contract/lib/contract/constructorMethods.js#L87

You are right that it mutates this (which is awkward), so you don't need the assigned value. I'd still argue it's cleaner to use the assigned value in case this changes in the future.